core/src/main/scala/sbt/librarymanagement/ResolverExtra.scala

```diff
@@ -391,6 +391,17 @@ private[librarymanagement] abstract class ResolverFunctions {
   def defaultRetrievePattern =
     "[type]s/[organisation]/[module]/" + PluginPattern + "[artifact](-[revision])(-[classifier]).[ext]"
   final val PluginPattern = "(scala_[scalaVersion]/)(sbt_[sbtVersion]/)"
+  private[librarymanagement] def expandMavenSettings(str: String): String = {
```

Comment:
Rather than scoping that to `mavenLocalDir`, I exposed it for testing purposes. Happy to backpedal on that, it doesn't seem to be the usual coding style of this project, but I felt tests would be useful for anything that involves regular expresses.